### PR TITLE
Put listings into the shared workspace

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Otherwise rust analyzer is very sad.
Also put numbered listings in own folder.
And changed references and CI accordingly

### CODE REVIEW

File: book/listings/Cargo.toml [hunk 1]
```diff
@@ -1,28 +1,116 @@
...
-    "widgets_1",
-    "widgets_2",
-]
+[package]
+name = "gtk4-rs-book-listings"
+version = "0.1.0"
+edition = "2018"
+
+[dependencies]
+once_cell = "1.5"
```
Comment (sdroege) [inline book/listings/Cargo.toml:7]: Why 1.5? 1.0 is probably enough? :)

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: book/listings/Cargo.toml (hunk 1 above)
Comment (Hofer-Julian) [reply book/listings/Cargo.toml:7]: Copy-pasta from examples :)

File: .github/workflows/book.yml [hunk 2]
```diff
@@ -20,9 +20,7 @@ jobs:
           toolchain: stable
           override: true
 
-      - run: cargo build
-        working-directory: book/listings
-
+      - run: cargo build --manifest-path book/listings/Cargo.toml --bins
```
Comment (bilelmoussaoui) [inline .github/workflows/book.yml:23]: What's the difference? working-directory does exactly what you need here.

**Comment by bilelmoussaoui** [comment]
otherwise lgtm

File: .github/workflows/book.yml (hunk 2 above)
Comment (Hofer-Julian) [reply .github/workflows/book.yml]: Good point, now that you say it :P
Fixed!

File: book/listings/Cargo.toml (hunk 1 above)
Comment (Hofer-Julian) [reply book/listings/Cargo.toml]: Fixed.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**